src/client/remote_model.py

```diff
@@ -66,13 +67,39 @@ def set_requires_grad(self, value):
         for p in self.parameters():
             p.requires_grad = value
 
-    def forward(self, *args, use_cache=None, **kwargs):
-        if use_cache:
-            raise ValueError(
-                "Distributed forward does not support use_cache; for efficient cache-aware generation, "
-                "please use model.transformer.inference_session() or model.generate(...)"
-            )
-        return super().forward(*args, use_cache=False, **kwargs)
+    def forward(
+        self, 
+        input_ids: Optional[torch.LongTensor] = None, 
+        inputs_embeds: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        **kwargs
+    ):
+        if input_ids is not None and inputs_embeds is not None:
```

Comment:
Consider adding a warning:

if kwargs:
  logger.warning("The following keyword arguments to DistributedBloom will will be ignored: {kwargs}")